.github/workflows/workflow-lint.yml

```diff
@@ -0,0 +1,80 @@
+name: lint
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+
+jobs:
+  # Runs golangci to ensure that:
+  # 1. the tooling is working on the target platform
+  # 2. the linter is happy
+  # 3. for canary (if there is a canary go version), does lint for all supported goos
+  lint-go:
+    name: "go${{ inputs.hack }}"
+    uses: ./.github/workflows/job-lint-go.yml
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - runner: ubuntu-24.04
+            goos: linux
+          - runner: ubuntu-24.04
+            goos: freebsd
+          - runner: macos-15
+            goos: darwin
+          # FIXME: this is currently failing in a nonsensical way, so, running on linux instead...
+          # - runner: windows-2022
+          - runner: ubuntu-24.04
+            goos: windows
+          # Additionally lint for canary
+          - runner: ubuntu-24.04
+            goos: linux
+            canary: true
+    with:
+      timeout: 5
+      go-version: "1.24"
+      runner: ubuntu-24.04
+      # Note: in GitHub yaml world, if `matrix.canary` is undefined, and is passed to `inputs.canary`, the job
+      # will not run. However, if you test it, it will coerce to `false`, hence:
```

Comment:
😄 
Maybe this should be reported as a bug? (in https://github.com/actions/toolkit/issues ?)